### DISCUSSION TITLE
CLI and Application Version

### DISCUSSION THREAD

**Discussion by paganotoni** [gobuffalo/cli#143] [General]
One thing that has somehow confused Buffalo developers is our choice to separate the two codebases, the CLI and the Buffalo Library. 

While I think this was the right way of organizing the codebase I wanted to surface this topic as some developers have shown their confusion with the version numbers and maybe we can come up with ways to keep everybody informed about those 2 versions and releases.

cc @gobuffalo/all

**Comment by sio4** [comment]
While I also think the codebase separation is the right decision, I agree that that versioning is very confusing. However, in my opinion, the reason for the confusion does not come from the version itself but from the names of those repositories. For buffalo users, especially those new to buffalo, "buffalo" may means the command `buffalo`.

I feel like the ideal names are (if I name them without consideration of any existing things) `gobuffalo/buffalo` for the current `gobuffalo/cli` project and `gobuffalo/core` or `gobuffalo/engine` or something like `gobuffalo/buffalo-core` for the current `gobuffalo/buffalo` project. (but just exposed as `package buffalo`)

Actually, when I heard that Mark secured the Github organization `buffalo`, what I think for the first time was the name change. :-)

**Reply by sio4** [reply]
One other direction is that, if we modulized the cli enough, with cli plugins and generators, so if the core of the cli became light and there are not many things to be changed, I think we can merge them again to reduce the confusion (while hardening the user experiences of cli/dev-cycle by improving plugins/generators)
